lib/std/fs/File.zig

```diff
@@ -1617,14 +1617,23 @@ pub const LockError = error{
 /// A process may hold only one type of lock (shared or exclusive) on
 /// a file. When a process terminates in any way, the lock is released.
 ///
-/// Assumes the file is unlocked.
+/// Assumes the file is unlocked, or in the case of Lock.none, locked.
 ///
 /// TODO: integrate with async I/O
 pub fn lock(file: File, l: Lock) LockError!void {
     if (is_windows) {
         var io_status_block: windows.IO_STATUS_BLOCK = undefined;
         const exclusive = switch (l) {
-            .none => return,
+            .none => return windows.UnlockFile(
+                file.handle,
+                &io_status_block,
+                &range_off,
+                &range_len,
+                null,
+            ) catch |err| switch (err) {
+                error.RangeNotLocked => unreachable, // The file is assumed to be locked.
+                error.Unexpected => unreachable, // Resource deallocation must succeed.
```

Comment:
Indeed resource deallocation must succeed, but you can't use `unreachable` here unless you have evidence that Windows not only will never return an error code from UnlockFile but also won't ship an OS update in the future that makes this function start returning an error code under some conditions.

Reply:
that makes sense, would a panic be correct here?

Reply:
It looks like this is the current behavior in `File.unlock` as well, https://github.com/ziglang/zig/blob/4d09fb491f1a2e3c4404807c5f4b2c2e114f911b/lib/std/fs/File.zig#L1670, which, while not part of my initial scope, I would not be opposed to updating as well. What is the proper method for dealing with a system deallocation function that may fail?

Reply:
```suggestion
                else => |e| return e,
```

The errors are handled with `unreachable` in `unlock` because it can't return an error, but `File.lock` can, so just returning the error would make the most sense.

The same change could be made in `File.downgradeLock` which has the same
```zig
error.Unexpected => unreachable, // Resource deallocation must succeed.
```
logic.

Reply:
I have updated the required code.